src/workflows/testing/runner.py

```diff
@@ -0,0 +1,91 @@
+import inspect
+
+from typing import AsyncContextManager, Any, Callable, Optional, TYPE_CHECKING
+from types import TracebackType
+from collections import Counter
+
+if TYPE_CHECKING:
+    from workflows import Workflow
+from workflows.events import StartEvent, Event, EventType
+
+
+class WorkflowTestRunner(AsyncContextManager["WorkflowTestRunner"]):
+    def __init__(self, workflow: "Workflow"):
+        self._workflow = workflow
+
+    async def __aenter__(self) -> Any:
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> Optional[bool]:
+        """Called when exiting the 'async with' block"""
+        return not exc_type and not exc_val and not exc_tb
+
+    async def run_and_collect(
```

Comment:
I would call this just `run`, the "collection" is implicit in the fact that this is a test runner